libs/barista-components/table/src/table.ts

```diff
@@ -321,4 +332,41 @@ export class DtTable<T> extends _DtTableBase<T> implements OnDestroy {
 
   /** CSS class added to any row or cell that has sticky positioning applied. */
   protected stickyCssClass = 'dt-table-sticky';
+
+  exportCSV(): void {
+    let ds = this.dataSource as DtTableDataSource<T>;
+    let exportData = ds.filteredData;
+    let csv = '';
+    let keys: string[] = [];
+    //get list of keys
+    if (typeof exportData[0] == 'object') {
+      keys = Object.keys(exportData[0]);
+    }
+    if (!keys.length) return;
+    csv += keys.join(',') + '\n';
+
+    exportData.forEach((row) => {
+      let vals = Object.values(row);
+      keys.forEach((key, idx) => {
+        let val = row[key];
+        if (typeof val != 'string') val = val.toString();
```

Comment:
@LucasHocker The datasource can also hold complex values, not only primitives like the ones in your example. Data structures like this are not uncommon. 
```
 {
      host: 'et-demo-2-win4',
      cpu: 30,
      memory: {
        memoryPerc: 38,
        memoryTotal: 5830000000,
      },
      traffic: 98700000,
    },
  ...
  ```

With your implementation this would very likely result in a [Object object] stringification. I think we need another iteration on this part of the logic to get this rendering correctly.